File: TcpDemo/app/src/main/java/edu/cs4730/tcpdemo/MainFragment.java
package edu.cs4730.tcpdemo;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import edu.cs4730.tcpdemo.databinding.FragmentMainBinding;

public class MainFragment extends Fragment {

    FragmentMainBinding binding;
    String TAG = "MainFragment";

    public MainFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = FragmentMainBinding.inflate(inflater, container, false);

        binding.helpClient.setOnClickListener(Navigation.createNavigateOnClickListener(R.id.action_help_to_client, null));
        binding.helpServer.setOnClickListener(Navigation.createNavigateOnClickListener(R.id.action_help_to_server, null));

        return binding.getRoot();

    }

    //A simple method to append data to the logger textview.
    public void logthis(String msg) {
        binding.logger.append(msg + "\n");
        Log.d(TAG, msg);
    }
}

File: TcpDemo_kt/app/src/main/java/edu/cs4730/tcpdemo_kt/MainFragment.kt
package edu.cs4730.tcpdemo_kt

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation.createNavigateOnClickListener
import edu.cs4730.tcpdemo_kt.databinding.FragmentMainBinding

class MainFragment : Fragment() {
    private lateinit var binding: FragmentMainBinding
    private var TAG: String = "MainFragment"

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        binding = FragmentMainBinding.inflate(inflater, container, false)

        binding.helpClient.setOnClickListener(
            createNavigateOnClickListener(
                R.id.action_help_to_client, null
            )
        )
        binding.helpServer.setOnClickListener(
            createNavigateOnClickListener(
                R.id.action_help_to_server, null
            )
        )

        return binding.root
    }

    //A simple method to append data to the logger textview.
    fun logthis(msg: String) {
        binding.logger.append(msg + "\n")
        Log.d(TAG, msg)
    }
}